### PR TITLE
Populate shipments form

### PR DESCRIPTION
**Pull request by Levia**
This is about [completing the shipments edit form](https://www.pivotaltracker.com/story/show/135324665).
The dropdowns have been populated moving code from SAPI regarding the models, but also using the SAPI api regarding autocompletion of the taxon name.
To achieve this, select2 has been used.

Still regarding the taxon name, at the moment I'm not showing the name status along with the taxon name, but it's indeed possibile by, for example, uncommenting the line ` #visibility: 'trade_internal' #includes status` in the ajax call.

Originally, by Michel's design, the quantity field was a dropdown, but it seems to me that this should have been a normal text input, so I've also changed this.

### CODE REVIEW

File: app/views/shipments/_form.html.erb [hunk 1]
```diff
@@ -33,21 +33,20 @@
         title: 'Term',
         name: 'term_code',
         placeholder: "Please select appendix value",
-        data: ['a', 'b', 'c'],
+        data: Term.pluck(:code),
```
Comment (agnessa) [inline app/views/shipments/_form.html.erb]: can we apply explicit alphabetic ordering for all trade codes and geo entities drop downs?

File: app/views/shipments/_form.html.erb [hunk 2]
```diff
@@ -18,10 +18,10 @@
         title: 'Taxon name',
         name: 'taxon_name',
         placeholder: "Please select appendix value",
-        data: ['a', 'b', 'c'],
         enabled: true,
         value: (@shipment && @shipment.taxon_name),
-        form: (form && form.object_name)
+        form: (form && form.object_name),
+        apiBaseUrl: Rails.application.secrets.sapi_staging_path
```
Comment (agnessa) [inline app/views/shipments/_form.html.erb]: I don't think I have this setting in secrets.yml? if we need to add it, let's make sure there's no "staging" in the name, we can just call it sapi_path and set separately for each env. Also, if there's anything required in the .env file as well, please add it to .env.example

**Comment by agnessa** [comment]
When we edit an individual shipment, the button should read just "Update" rather than "Update selection"

**Comment by Levia** [comment]
Thank you for your comments! I've made those amendments, so it' ready again.
The only question I have is if we want to display the full name of the country rather then the iso code when that field is already populated.

**Comment by agnessa** [comment]
We want to display whatever is physically in the sandbox, so if it was populated with "ZONK" we display "ZONK", and therefore I think the iso code is more appropriate to show when you first load the form. But it does look strange that we have full country names in the dropdown and then iso code when you first load. The simple way out of this is just show iso codes in dropdown, and maybe that's good enough to do?

**Comment by Levia** [comment]
Yes we can do that. I replicated the way the current tool in SAPI works and it seemed to be easier for the user to search for the name of the country rather then the iso code.
But if they have full knowledge of the iso code when they are doing the amendments, then it's fine to me 😅

**Pull request merged**